packages/nextra-theme/src/index.tsx

```diff
@@ -30,10 +28,118 @@ import {
 } from '@/components'
 import { DocumentContext, MDXLayoutNav, MDXLayoutOutline, MDXLayoutPagination, NavContext } from '@/layout'
 
+const MDXWrapper: NextraMDXContent = ({ children, toc }) => {
```

Comment:
Am I correct that this refactoring (turning this into a component that is passed to `MDXProvider`) is needed because `pageOpts` no longer includes `headings`? (Or does it still include it, but you felt that this was better for some reason?)

Reply:
this refactoring is needed because `headings` (or `toc` in v3) now passed only in `wrapper` component

mdx docs about wrapper
<img width="882" alt="image" src="https://github.com/user-attachments/assets/f607af02-817e-45a5-a50b-db775de62689">

or more in my blogpost https://the-guild.dev/blog/nextra-3#pageoptsheadings-was-removed